File: amss/lib/models/boardgame.dart
class Boardgame {
  int id = -1;
  String name = "";
  int minimumPlayers = -1;
  int maximumPlayers = -1;
  int minimumAgeRequirement = -1;
  int estimatedPlayTime = -1;
  String htmlRules = "";

  static Boardgame fromMap(Map<String, dynamic> json) {
    final bg = Boardgame();
    bg.id = json["id"];
    bg.name = json["name"];
    bg.minimumPlayers = json["minimumPlayers"];
    bg.maximumPlayers = json["maximumPlayers"];
    bg.minimumAgeRequirement = json["minimumAgeRequirement"];
    bg.estimatedPlayTime = json["estimatedPlayTime"];
    bg.htmlRules = json["htmlRules"];
    return bg;
  }
}

File: backend/src/main/java/com/example/backend/models/BoardGame.java
package com.example.backend.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity(name = "boardgames")
@Table(
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"id"}),
                @UniqueConstraint(columnNames = {"name"}),
        }
)
@AllArgsConstructor
@NoArgsConstructor
public class BoardGame {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Getter
    public Long id;

    public String name;
    public Integer minimumPlayers;
    public Integer maximumPlayers;
    public Integer minimumAgeRequirement;
    public Integer estimatedPlayTime;
    @Column(columnDefinition="TEXT", length = 102020)
    public String htmlRules;
}
